### ISSUE TITLE
ttnn.repeat generates a tensor with incorrect shape

### ISSUE THREAD

**Issue by kevinwuTT** [tenstorrent/tt-metal#11168]
**Describe the bug**
If I have an input tensor with a shape of `(1, 1, 1, 1)` and call `ttnn.repeat` with a shape of `(2, 2, 2, 2)`, the output shape doesn't appear correct. The output shape should be `(2, 2, 2, 2)`, but ttnn output shape is `(2, 2, 33, 33)` instead. 

If I change the layout to row major, I instead get an error:

```
RuntimeError: TT_FATAL @ ../tt_metal/impl/buffers/buffer.cpp:31: size != 0 and page_size != 0
info:
Buffer size and page size should be larger than 0 bytes!
```

**To Reproduce**
```
import ttnn
import torch

with ttnn.manage_device(device_id=0) as device:
    torch_input_tensor = torch.randn((1, 1, 1, 1), dtype=torch.bfloat16)
    repeat_shape = (2, 2, 2, 2)
    
    torch_result = torch_input_tensor.repeat(repeat_shape)
    
    input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT, device=device)
    output = ttnn.repeat(input_tensor, ttnn.Shape(repeat_shape))
    output = ttnn.to_torch(output)

    print("torch shape:", torch_result.shape)
    print("ttnn shape:", output.shape)
```
Output:
```
torch shape: torch.Size([2, 2, 2, 2])
ttnn shape: torch.Size([2, 2, 33, 33])
```

**Expected behavior**
ttnn shape should be `(2, 2, 2, 2)` in this example.

**System**
- OS: Ubuntu 20.04.6 LTS
- tt-metal commit: bc0a96a8

**Comment by jaykru-tt** [comment]
Added fixes for this shape in #13507. Let me know if you're still having issues.

**Comment by jaykru-tt** [comment]
Going to close this now. @kevinwuTT reopen this issue if you have any further problems with shapes of this form :)
